listener/kcp/reconciler_factory.go

```diff
@@ -107,14 +109,32 @@ func PreReconcile(
 	cr *apischema.CRDResolver,
 	io workspacefile.IOHandler,
 ) error {
-	JSON, err := cr.Resolve()
+	actualJSON, err := cr.Resolve()
 	if err != nil {
+		// If we cannot resolve the schema, we return an error.
```

Comment:
This comment doesn't hold any additional sense, we can derive the idea from the code.
Lets avoid comments that are explaining simple things.